mlflow/sklearn/utils.py

```diff
@@ -103,6 +118,248 @@ def _get_args_for_score(score_func, fit_func, fit_args, fit_kwargs):
+                    ),
+                ),
+            ]
+        )
+
+    return _get_metrics_value_dict(classifier_metrics)
+
+
+def _get_regressor_metrics(trained_estimator, fit_args, fit_kwargs):
+    """
+    Compute and log various common metrics for regressors
+
+    For (1) (root) mean squared error:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_error.html#sklearn.metrics.mean_squared_error
+    (2) mean absolute error:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_absolute_error.html#sklearn.metrics.mean_absolute_error
+    (3) r2 score:
+    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html#sklearn.metrics.r2_score
+    By default, we choose the parameter `multioutput` to be `uniform_average`
+    to average outputs with uniform weight.
+
+    Steps:
+    1. Extract X and y_true from fit_args and fit_kwargs, and compute y_pred.
+    2. If the sample_weight argument exists in fit_func (accuracy_score by default
+    has sample_weight), extract it from fit_args or fit_kwargs as
+    (y_true, y_pred, sample_weight, multioutput), otherwise as (y_true, y_pred, multioutput)
+    3. return a dictionary of metric(name, value)
+
+    :param trained_estimator: The already fitted regressor
+    :param fit_args: Positional arguments given to fit_func.
+    :param fit_kwargs: Keyword arguments given to fit_func.
+    :return: dictionary of (function name, computed value)
+    """
+    import sklearn
+
+    fit_arg_names = _get_arg_names(trained_estimator.fit)
+    _, y_true, y_pred = _get_samples_labels_and_predictions(
+        trained_estimator, fit_args, fit_kwargs, fit_arg_names
+    )
+    sample_weight = (
+        _get_sample_weight(fit_arg_names, fit_args, fit_kwargs)
+        if _SAMPLE_WEIGHT in fit_arg_names
+        else None
+    )
+
+    regressor_metrics = [
+        _SklearnMetric(
+            name="mse",
+            function=sklearn.metrics.mean_squared_error,
+            arguments=dict(
+                y_true=y_true,
+                y_pred=y_pred,
+                sample_weight=sample_weight,
+                multioutput="uniform_average",
+            ),
+        ),
+        _SklearnMetric(
+            name="mae",
+            function=sklearn.metrics.mean_absolute_error,
+            arguments=dict(
+                y_true=y_true,
+                y_pred=y_pred,
+                sample_weight=sample_weight,
+                multioutput="uniform_average",
+            ),
+        ),
+        _SklearnMetric(
+            name="r2_score",
+            function=sklearn.metrics.r2_score,
+            arguments=dict(
+                y_true=y_true,
+                y_pred=y_pred,
+                sample_weight=sample_weight,
+                multioutput="uniform_average",
+            ),
+        ),
+    ]
+
+    # To be compatibale with deprecated versions of scikit-learn (below 0.22.2), where
```

Comment:
Nit: small typo in "compatible". Also, these are older versions of sklearn, not necessarily deprecated versions